docs/2024/puzzles/day13.md

```diff
@@ -6,10 +6,208 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 https://adventofcode.com/2024/day/13
 
+## Solution Summary
+
+This problem requires you to work with equations and numbers.
+
+Both parts of the problem asks you to calculate the smallest number of tokens needed to win as many prizes as possible, which means to calculate the optimal number of A and B button presses.
+
+- For part 1, the question assures that buttons are pressed no more than 100 times, so it can be simply brute forced for all possible combinations.
+- For part 2, the position of prize is higher by 10000000000000, which means brute-force approach is not feasible. We need to use the power of math to calculate the optimal solution.
+  - First, describe the relation between button presses and prize position can as a multivariable equation.
+  - Then, use the algebraic properties to simplify the equation.
+  - This way, we reduce the complexity of the problem to `O(1)`.
+
+## Parsing
+
+The input data is quite complex to parse. First, let's create a case class to store the amount of claw movements and the prize position:
+
+```scala
+case class Claw(ax: Long, ay: Long, bx: Long, by: Long, x: Long, y: Long)
+```
+
+Since a lot of numbers needs to be parsed, [Extractor Objects](https://docs.scala-lang.org/tour/extractor-objects.html) come in handy. this `L` object with an `unapply` method will parse a string to `Long`:
```

Comment:
```suggestion
Since a lot of numbers need to be parsed, [Extractor Objects](https://docs.scala-lang.org/tour/extractor-objects.html) come in handy. this `L` object with an `unapply` method will parse a string to `Long`:
```